C++ usage

```
In file included from /home/bernhard/Projects/nello_firmware_ng/src/main.cpp:4:0:
/home/bernhard/Projects/zephyr/include/sensor.h: In function 'int sensor_attr_set(device*, sensor_channel, sensor_attribute, const sensor_value*)':
/home/bernhard/Projects/zephyr/include/sensor.h:271:45: error: invalid conversion from 'const void*' to 'const sensor_driver_api*' [-fpermissive]
  const struct sensor_driver_api *api = dev->driver_api;
                                        ~~~~~^~~~~~~~~~
/home/bernhard/Projects/zephyr/include/sensor.h: In function 'int sensor_trigger_set(device*, sensor_trigger*, sensor_trigger_handler_t)':
/home/bernhard/Projects/zephyr/include/sensor.h:299:45: error: invalid conversion from 'const void*' to 'const sensor_driver_api*' [-fpermissive]
  const struct sensor_driver_api *api = dev->driver_api;
```

my config includes the `CONFIG_CPLUSPLUS=y` the target is the nrf52 dk dev boad.

How do I get around this? Specifiying `-fpermissive` does not solve my issue. I tried to include all C includes in `extern "C"` but no success.

---

`CONFIG_COMPILER_OPT="-fpermissive"` does not help either

--- a/include/drivers/sensor.h
+++ b/include/drivers/sensor.h
@@ -345,7 +345,8 @@ static inline int z_impl_sensor_attr_set(struct device *dev,
 					enum sensor_attribute attr,
 					const struct sensor_value *val)
 {
-	const struct sensor_driver_api *api = dev->driver_api;
+	const struct sensor_driver_api *api =
+		(const struct sensor_driver_api *)dev->driver_api;
 
 	if (api->attr_set == NULL) {
 		return -ENOTSUP;
@@ -375,7 +376,8 @@ static inline int sensor_trigger_set(struct device *dev,
 				     struct sensor_trigger *trig,
 				     sensor_trigger_handler_t handler)
 {
-	const struct sensor_driver_api *api = dev->driver_api;
+	const struct sensor_driver_api *api =
+		(const struct sensor_driver_api *)dev->driver_api;
 
 	if (api->trigger_set == NULL) {
 		return -ENOTSUP;
@@ -404,7 +406,8 @@ __syscall int sensor_sample_fetch(struct device *dev);
 
 static inline int z_impl_sensor_sample_fetch(struct device *dev)
 {
-	const struct sensor_driver_api *api = dev->driver_api;
+	const struct sensor_driver_api *api =
+		(const struct sensor_driver_api *)dev->driver_api;
 
 	return api->sample_fetch(dev, SENSOR_CHAN_ALL);
 }
@@ -434,7 +437,8 @@ __syscall int sensor_sample_fetch_chan(struct device *dev,
 static inline int z_impl_sensor_sample_fetch_chan(struct device *dev,
 						 enum sensor_channel type)
 {
-	const struct sensor_driver_api *api = dev->driver_api;
+	const struct sensor_driver_api *api =
+		(const struct sensor_driver_api *)dev->driver_api;
 
 	return api->sample_fetch(dev, type);
 }
@@ -468,7 +472,8 @@ static inline int z_impl_sensor_channel_get(struct device *dev,
 					   enum sensor_channel chan,
 					   struct sensor_value *val)
 {
-	const struct sensor_driver_api *api = dev->driver_api;
+	const struct sensor_driver_api *api =
+		(const struct sensor_driver_api *)dev->driver_api;
 
 	return api->channel_get(dev, chan, val);
 }

